run_windows.go

```diff
@@ -0,0 +1,527 @@
+			return nil, err
+		}
+	}
+
+	// unlike unix command exec, do not set stdin/stdout/stderr
+
+	c.SysProcAttr = attrs
+
+	// do not use os/exec.Start since we need to append console handler to startup info
+
+	err = start((*cmd)(unsafe.Pointer(c)), syscall.Handle(tty.Fd()))
+	if err != nil {
+		return nil, err
+	}
+
+	return pty, err
+}
+
+func createExtendedStartupInfo(consoleHandle syscall.Handle) (_ *startupInfoEx, err error) {
+	// append console handler to new process
+	var (
+		attrBufSize uint64
+		si          startupInfoEx
+	)
+
+	si.startupInfo.Cb = uint32(unsafe.Sizeof(si))
+
+	// get size of attr list
+	r1, _, err := initializeProcThreadAttributeList.Call(
+		0, // list ptr
+		1, // list item count
+		0, // dwFlags: reserved, MUST be 0
+		uintptr(unsafe.Pointer(&attrBufSize)),
+	)
+	if r1 == 0 {
+		// according to
+		// https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-initializeprocthreadattributelist
+		// which says: This initial call will return an error by design. This is expected behavior.
+		//
+		// so here we check the returned value of the attr buf size, if it's zero, we cannot update attribute list
+		if attrBufSize == 0 {
+			return nil, os.NewSyscallError("InitializeProcThreadAttributeList (size)", err)
+		}
+	}
+
+	attrListBuf := make([]byte, attrBufSize)
+	si.lpAttrList = syscall.Handle(unsafe.Pointer(&attrListBuf[0]))
+	// create attr list with console handler
+	r1, _, err = initializeProcThreadAttributeList.Call(
+		uintptr(si.lpAttrList),                // attr list buf
+		1,                                     // list item count
+		0,                                     // dwFlags: reserved, MUST be 0
+		uintptr(unsafe.Pointer(&attrBufSize)), // size of the list
+	)
+	if r1 == 0 {
+		// false
+		return nil, os.NewSyscallError("InitializeProcThreadAttributeList (create)", err)
+	}
+
+	// https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-updateprocthreadattribute
+	r1, _, err = updateProcThreadAttribute.Call(
+		uintptr(si.lpAttrList), // buf list
+		0,                      // dwFlags: reserved, MUST be 0
+		_PROC_THREAD_ATTRIBUTE_PSEUDOCONSOLE,
+		uintptr(consoleHandle),
+		unsafe.Sizeof(consoleHandle),
+		0,
+		0,
+	)
+	if r1 == 0 {
+		// false
+		_, _, _ = deleteProcThreadAttributeList.Call(uintptr(si.lpAttrList))
+		return nil, os.NewSyscallError("UpdateProcThreadAttribute", err)
+	}
+
+	return &si, nil
+}
+
+// copied from os/exec.(*Cmd).Start
+// start starts the specified command but does not wait for it to complete.
+//
+// If Start returns successfully, the c.Process field will be set.
+//
+// The Wait method will return the exit code and release associated resources
+// once the command exits.
+func start(c *cmd, consoleHandle syscall.Handle) error {
+	if c.lookPathErr != nil {
+		c.closeDescriptors(c.closeAfterStart)
+		c.closeDescriptors(c.closeAfterWait)
+		return c.lookPathErr
+	}
+	if runtime.GOOS == "windows" {
```

Comment:
Is this needed? We are already in a `_windows.go` file, so it will only be compiled for windows. Can the binary run with GOOS != "windows"?

Reply:
This was copied from the standard package as function comment suggested. I was trying to keep them as is, so when we upgrade this part we can know where changed exactly.